### PR TITLE
Add FERC1 output table that combines key FERC1 subtables

### PR DESCRIPTION
**Pull request by aesharpe**
This PR contains the code needed to create a composite FERC1 output table containing the information from the FERC1 `steam`, `small_gens`, `hydro`, and `pumped_storage` tables--collectively representing all of the plants reporting to FERC.

This table may have multiple uses, but the main impetus was for mapping FERC records to EIA records via the Master Unit List and FERC-EIA mapping established in the RMI repo.

Besides combining the subtables, this PR also adds new columns (like `capacity_factor` and in one case `opex_fuel_per_mwh`) to the subtables where possible so that they may be more accurately matched to EIA records. These columns are only added to subtables when they can be calculated from existing columns. 

My main question -- and why this is a _draft_ PR vs a real PR -- is whether this information belongs in the `output` module or the `analysis` module. While I originally thought it fit best among the outputs, I recognize that I have added some calculated fields more akin to the analysis. I would love to know what you think is most appropriate!

### CODE REVIEW

**Review by cmgosnell — CHANGES_REQUESTED** [review]
It definitely makes sense to me to keep this table in the outputs, not in analysis.

It'd be good to have this output run in the ci tests!

File: src/pudl/output/ferc1.py [hunk 1]
```diff
@@ -148,8 +152,12 @@ def plants_small_ferc1(pudl_engine):
     plants_small_df = (
         pd.read_sql_table("plants_small_ferc1", pudl_engine)
         .drop(['id'], axis="columns")
-        .merge(pd.read_sql_table("utilities_ferc1", pudl_engine),
-               on="utility_id_ferc1")
+        .merge(plants_utils_ferc1(pudl_engine),
+               on=['utility_id_ferc1', 'plant_name_ferc1'])
+        .assign(
+            capacity_factor=lambda x:
+                x.net_generation_mwh / (8760 * x.capacity_mw),
```
[the remark this answers is not in the document]
Comment (aesharpe) [reply src/pudl/output/ferc1.py:159]: Can you point me to where that helper function is?

File: src/pudl/output/ferc1.py [hunk 2]
```diff
@@ -226,3 +238,77 @@ def plant_in_service_ferc1(pudl_engine):
...
+    logger.info("loading small gens table")
+    small_df = plants_small_ferc1(pudl_engine)
+    logger.info("loading hydro table")
+    hydro_df = plants_hydro_ferc1(pudl_engine)
+    logger.info("loading pumped storage table")
+    pump_df = plants_pumped_storage_ferc1(pudl_engine)
+    logger.info("loading fbp table")
+    fbp_df = fuel_by_plant_ferc1(pudl_engine)
+
+    # Prep steam table (merge with fbp)
```
[the remark this answers is not in the document]
Comment (aesharpe) [reply src/pudl/output/ferc1.py:264]: hmm that's a fair point! I'll do it for now.

File: src/pudl/output/ferc1.py [hunk 3]
```diff
@@ -226,3 +238,77 @@ def plant_in_service_ferc1(pudl_engine):
...
+        .rename(columns={'opex_plants': 'opex_plant'})
+        .pipe(pudl.helpers.convert_cols_dtypes,
+              'ferc1', 'ferc1 plant records'))
+
+    if 0.9 > (len(steam_fbp_df) /
+              len(steam_fbp_df.drop_duplicates(
+                  subset=['report_year',
+                          'utility_id_pudl',
+                          'plant_id_ferc1'])) < 1.1):
+        raise AssertionError(
```
[the remark this answers is not in the document]
Comment (aesharpe) [reply src/pudl/output/ferc1.py:288]: you mean something like `pd.merge(steam, fbp, how='left', on=....)` and get rid of the assertion and if len() < stuff?

File: src/pudl/output/ferc1.py [hunk 4]
```diff
@@ -226,3 +238,77 @@ def plant_in_service_ferc1(pudl_engine):
...
+    # Combine all the tables together
+    logger.info("combining all tables")
+    all_plants_df = (
+        pd.concat([steam_fbp_df, small_df, hydro_df, pump_df])
+        .rename(columns={
+            'fuel_cost': 'total_fuel_cost',
+            'fuel_mmbtu': 'total_mmbtu',
+            'opex_fuel_per_mwh': 'fuel_cost_per_mwh',
+            'primary_fuel_by_mmbtu': 'fuel_type_code_pudl',
+            'record_id': 'record_id_ferc1'})
```
[the remark this answers is not in the document]
Comment (aesharpe) [reply src/pudl/output/ferc1.py:305]: mmm ok I think it's fine to leave it as `record_id`

File: src/pudl/output/ferc1.py [hunk 5]
```diff
@@ -226,3 +238,77 @@ def plant_in_service_ferc1(pudl_engine):
...
+    logger.info("prepping hydro tables")
+    hydro_df = hydro_df.rename(columns={'project_num': 'ferc_license_id'})
+    pump_df = pump_df.rename(columns={'project_num': 'ferc_license_id'})
+
+    # Combine all the tables together
+    logger.info("combining all tables")
+    all_plants_df = (
+        pd.concat([steam_fbp_df, small_df, hydro_df, pump_df])
+        .rename(columns={
+            'fuel_cost': 'total_fuel_cost',
```
[the remark this answers is not in the document]
Comment (aesharpe) [reply src/pudl/output/ferc1.py:301]: probably!!

File: src/pudl/output/ferc1.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (aesharpe) [reply src/pudl/output/ferc1.py:288]: I will play around with the fbp!

File: src/pudl/output/ferc1.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (aesharpe) [reply src/pudl/output/ferc1.py:159]: will just keep it as is for now b/c def annual

**Comment by aesharpe** [comment]
ADD THIS OUTPUT TO CI TEST

**Pull request merged**